pkg/utils/utils.go

```diff
@@ -91,10 +97,21 @@ func GetDeviceIDFromNetworkStatus(status netdefv1.NetworkStatus) (string, error)
 }
 
 // IPsFromStrings receives a list of IPs in string format and returns a list of net.IP
+// IPv4 IPs will have length of net.IPv4Len
+// IPv6 IPs will have length of net.IPv6Len
+// invalid IPs will be nil net.IP
 func IPsFromStrings(ips []string) []net.IP {
 	netIPs := make([]net.IP, 0, len(ips))
 	for _, ip := range ips {
-		netIPs = append(netIPs, net.ParseIP(ip))
+		// net.ParseIP() always returns a net.IP with length net.IPv6Len
+		// transform to IP version specific len
+		parsedIP := net.ParseIP(ip)
+		ipv4 := parsedIP.To4()
```

Comment:
I think this brakes semantic of the net package. In the net package ipv4 can be 4 or 16 byte. Your utils.IsIPv4 function can be safely used only with IPs from utils.IPsFromStrings, because utils.IsIPv4 may return false negative if valid ipv4 which represented as 16-byte passed.

I think it is better to revert change in IPsFromStrings function and instead refactor IsIPv4 function to use reliable check.

```
func IsIPv4(ip net.IP) bool {
	return ip.To4() != nil
}
```

Reply:
ack, i actually debated (with myself) the two approaches.
i was suprised that the length of an ipv4 address is 16 bytes, since its "in real life" 4 bytes.

however you are right, in any place where net.ParseIP() is used directly (or any other method that creates `IP` with fixed 16 byte len) will break.

will change IsIPv4 instead.